bin/pkg.ml

```diff
@@ -227,18 +270,22 @@ module Lock = struct
     (* Construct a list of thunks that will perform all the file IO side
        effects after performing validation so that if materializing any
        lockdir would fail then no side effect takes place. *)
-    let write_disk_list =
+    let write_disk_list, summary_pps =
       List.map per_context
-        ~f:(fun { Per_context.lock_dir_path; version_preference } ->
+        ~f:(fun { Per_context.lock_dir_path; version_preference; _ } ->
           let summary, lock_dir =
             Dune_pkg.Opam.solve_lock_dir ~version_preference ~repo_selection
               opam_file_map
           in
-          Console.print_user_message
-            (Dune_pkg.Opam.Summary.selected_packages_message summary
-               ~lock_dir_path);
-          Lock_dir.Write_disk.prepare ~lock_dir_path lock_dir)
+          let summary_message =
+            Dune_pkg.Opam.Summary.selected_packages_message summary
+              ~lock_dir_path
+            |> User_message.pp
+          in
+          (Lock_dir.Write_disk.prepare ~lock_dir_path lock_dir, summary_message))
+      |> List.split
     in
+    Format.printf "%a" Pp.to_fmt (Pp.concat ~sep:Pp.newline summary_pps);
```

Comment:
Should be `Dune_console.print` or the like. Otherwise this will break threading.

Reply:
Ah cool I didn't know about `Dune_console.print`. Updated.

Reply:
Or really `Console.printf`.

Reply:
Looks like `Console` is an alias of `Dune_console` (at least in the context of this line)? Should one be preferred over the other?

Reply:
The library is called `dune_console` we make an alias `Console` so we don't have to write `Dune_` everywhere.

Reply:
Gotcha. Updated.